src/server/game/Spells/SpellEffects.cpp

```diff
@@ -4392,23 +4392,37 @@ void Spell::EffectPullTowards(SpellEffIndex effIndex)
     if (!unitTarget)
         return;
 
+    float speedXY = m_spellInfo->Effects[effIndex].MiscValue / 10.0f;
+    float speedZ = damage / 10.0f;
     Position pos;
-    if (m_spellInfo->Effects[effIndex].Effect == SPELL_EFFECT_PULL_TOWARDS_DEST)
-    {
-        if (m_targets.HasDst())
-            pos.Relocate(*destTarget);
-        else
-            return;
-    }
-    else //if (m_spellInfo->Effects[i].Effect == SPELL_EFFECT_PULL_TOWARDS)
+    pos.Relocate(m_caster);
+
+    unitTarget->GetMotionMaster()->MoveJump(pos, speedXY, speedZ);
+}
+
+void Spell::EffectPullTowardsDest(SpellEffIndex effIndex)
+{
+    if (effectHandleMode != SPELL_EFFECT_HANDLE_HIT_TARGET)
+        return;
+
+    if (!unitTarget)
+        return;
+
+    if (!m_targets.HasDst())
     {
-        pos.Relocate(m_caster);
+        TC_LOG_ERROR("spells", "Spell %u with SPELL_EFFECT_PULL_TOWARDS_DEST has no dest target", m_spellInfo->Id);
+        return;
     }
 
-    float speedXY = float(m_spellInfo->Effects[effIndex].MiscValue) * 0.1f;
-    float speedZ = unitTarget->GetDistance(pos) / speedXY * 0.5f * Movement::gravity;
+    Position const* pos = m_targets.GetDstPos();
+    // This is a blizzlike mistake: this should be 2D distance according to projectile motion formulas, but Blizzard erroneously used 3D distance
+    float distXY = unitTarget->GetExactDist(pos);
+    float distZ = pos->GetPositionZ() - unitTarget->GetPositionZ();
 
-    unitTarget->GetMotionMaster()->MoveJump(pos, speedXY, speedZ);
+    float speedXY = m_spellInfo->Effects[effIndex].MiscValue / 10.0f;
+    float speedZ = (2 * speedXY * speedXY * distZ + Movement::gravity * distXY * distXY) / (2 * speedXY * distXY);
+
+    unitTarget->JumpTo(speedXY, speedZ, pos);
```

Comment:
Hm, I see, TC doesn't have the parameter for this method that serves to specify the direction of the knockback. Here `pos` is being erroneously passed to the `bool forward` parameter.

Here it is:
Unit.h
```diff
-        void JumpTo(float speedXY, float speedZ, bool forward = true);
+        void JumpTo(float speedXY, float speedZ, bool forward = true, Position* dest = nullptr);
```

Unit.cpp
```diff
-void Unit::JumpTo(float speedXY, float speedZ, bool forward)
+void Unit::JumpTo(float speedXY, float speedZ, bool forward, Position* dest)
 {
     float angle = forward ? 0 : M_PI;
+    if (dest)
+        angle += GetRelativeAngle(dest);
+
     if (GetTypeId() == TYPEID_UNIT)
         GetMotionMaster()->MoveJumpTo(angle, speedXY, speedZ);
```

Reply:
No change

Reply:
Not surprising.
I don't have a slightest clue why it's not working for you. The code is very simple, I can't imagine what can it depend on that I'm not seeing.
I'll wait for someone else to test it and report their results.

Reply:
My bad I forget to apply the last commit of this PR, So this is the locale diff: https://pastebin.com/qrAEFR5W
And this is the result: https://imgur.com/i1M7cfP

Reply:
Yes, NOW you're missing this piece of code the I posted above. Obviously you must also change the call to `unitTarget->JumpTo(speedXY, speedZ, true, pos);`.

Reply:
I get an error: c:\trinitysource355\src\server\game\spells\spelleffects.cpp(4425): error C2661: 'Unit::JumpTo' : aucune fonction surchargée ne nécessite 4 arguments (cannot take 4 arguments)

Reply:
Then you must be wrong in saying
> So this is the locale diff: https://pastebin.com/qrAEFR5W

Because if you had changed Unit::JumpTo method as shown, you wouldn't be getting that error...

Reply:
Yes I revert it to see if it will change something or not.
After reapplying everything I get this error:
``` 
Erreur	C2664	'void Unit::JumpTo(WorldObject *,float,bool)' : impossible de convertir l'argument 4 de 'const Position *' en 'Position *'	game	c:\trinitysource355\src\server\game\spells\spelleffects.cpp	4425

Reply:
I revert it to see if it changes something.
After applying everything I get this error:
``` 
Erreur	C2664	'void Unit::JumpTo(WorldObject *,float,bool)' : impossible de convertir l'argument 4 de 'const Position *' en 'Position *'	game	c:\trinitysource355\src\server\game\spells\spelleffects.cpp	4425

Reply:
Change the parameter from `Position* dest` to `Position const* dest`. I'll edit the comment.